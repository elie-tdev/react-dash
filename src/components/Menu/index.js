export { default as MobileMenu } from './MobileMenu';
export { default as MobileMenuItem } from './MobileMenuItem';
