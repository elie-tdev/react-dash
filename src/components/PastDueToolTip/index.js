export { default as PastDueToolTip } from './PastDueToolTip';
