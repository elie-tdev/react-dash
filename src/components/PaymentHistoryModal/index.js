export { default as PaymentHistoryModal } from './PaymentHistoryModal';
