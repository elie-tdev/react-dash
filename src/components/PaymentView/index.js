export { default as PaymentView } from './PaymentView';
export { default as PaymentCurrent } from './PaymentCurrent';
export { default as PaymentPastDue } from './PaymentPastDue';
export { default as ScheduledPayment } from './ScheduledPayment';
export { default as NextPaymentView } from './NextPaymentView';
export { default as EnrollInAutoPay } from './EnrollInAutoPay';
export { default as NextPaymentCard } from './NextPaymentCard';
export { default as PastDueCard } from './PastDueCard';
export { default as AutoPayCard } from './AutoPayCard';
