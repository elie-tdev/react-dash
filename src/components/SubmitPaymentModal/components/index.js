export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
