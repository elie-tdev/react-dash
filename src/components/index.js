export * from './LoanGlance';
export * from './Button';
export * from './Header';
export * from './Footer';
export * from './LabelValueColumn';
export * from './LabelValueRow';
export * from './Loader';
export * from './CenteredScreenContent';
export * from './PaymentView';
export * from './Pagination';
export * from './Screen';
export * from './LoanSelect';
export * from './StatementGlance';
export * from './Menu';
export * from './PaymentHistoryModal';
export * from './PaymentStatus';
export * from './PaymentType';
export * from './SubmitPaymentModal';
export * from './GridContainer';
export * from './PastDueToolTip';
