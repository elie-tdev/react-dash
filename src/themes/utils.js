export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
