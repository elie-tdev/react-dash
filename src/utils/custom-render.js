import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
