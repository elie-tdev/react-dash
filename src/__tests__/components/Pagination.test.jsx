import React from 'react';

import { render } from '@/utils/custom-render';
import { Pagination } from '@/components';

describe('Pagination', () => {
  test('Renders', () => {
    const { container } = render(<Pagination />);
    expect(container).toMatchSnapshot();
  });
});
