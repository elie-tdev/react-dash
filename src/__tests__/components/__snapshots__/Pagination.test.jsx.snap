// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders 1`] = `
<div>
  <div
    class="makeStyles-root-1"
    data-testid="Pagination"
  >
    <button
      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded makeStyles-previousNext-3 Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiPaginationItem-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </button>
    <ul
      class="makeStyles-ul-2"
    >
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          aria-current="true"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded Mui-selected"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded"
          tabindex="0"
          type="button"
        >
          3
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded"
          tabindex="0"
          type="button"
        >
          4
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded"
          tabindex="0"
          type="button"
        >
          5
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <div
          class="makeStyles-moreContainer-6"
        >
          <div
            class="makeStyles-moreIconContainer-7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-moreIcon-8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="makeStyles-li-4"
        data-testid="PaginationItem"
      >
        <button
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page makeStyles-page-5 MuiPaginationItem-rounded"
          tabindex="0"
          type="button"
        >
          10
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
    <button
      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded makeStyles-previousNext-3"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiPaginationItem-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
